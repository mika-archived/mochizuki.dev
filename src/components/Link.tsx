import * as PropTypes from "prop-types";
import React from "react";

const Link: React.FC<React.AnchorHTMLAttributes<HTMLAnchorElement>> = props => {
  const { children } = props;

  return (
    <a {...props} target="_blank" rel="noopener noreferrer">
      {children}
    </a>
  );
};

Link.propTypes = {
  children: PropTypes.oneOfType([PropTypes.object, PropTypes.string]).isRequired
};

export default Link;
