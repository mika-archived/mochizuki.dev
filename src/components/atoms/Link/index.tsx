import React from "react";

type Props = {
  children: any;
  className?: string;
  href: string;
};

const Link: React.FC<Props> = ({ children, className, href }) => {
  if (href.startsWith("/")) {
    return (
      <a className={className} href={href}>
        {children}
      </a>
    );
  }

  return (
    <a className={className} href={href} rel="noreferrer noopener">
      {children}
    </a>
  );
};

export default Link;
