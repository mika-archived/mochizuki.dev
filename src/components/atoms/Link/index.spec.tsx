import "@testing-library/jest-dom/extend-expect";
import "jest-styled-components";

import React from "react";
import { cleanup, render } from "@testing-library/react";

import Link from ".";

afterEach(cleanup);

test("does not have changes from snapshot with internal link", () => {
  const { asFragment } = render(<Link href="/another">Another Link</Link>);

  expect(asFragment()).toMatchSnapshot();
});

test("does not have changes from snapshot with external link", () => {
  const { asFragment } = render(<Link href="https://example.com/another">Another Link</Link>);

  expect(asFragment()).toMatchSnapshot();
});
