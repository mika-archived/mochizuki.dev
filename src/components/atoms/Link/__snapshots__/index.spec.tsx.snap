// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not have changes from snapshot with external link 1`] = `
<DocumentFragment>
  <a
    href="https://example.com/another"
    rel="noreferrer noopener"
  >
    Another Link
  </a>
</DocumentFragment>
`;

exports[`does not have changes from snapshot with internal link 1`] = `
<DocumentFragment>
  <a
    href="/another"
  >
    Another Link
  </a>
</DocumentFragment>
`;
